polaris/viz/__init__.py

```diff
@@ -0,0 +1,138 @@
+from matplotlib.colors import LogNorm
+from matplotlib.patches import Polygon
+
+
+def plot_horiz_field(config, ds, dsMesh, fieldName, outFileName,
+                     title=None, tIndex=None, zIndex=None,
+                     vmin=None, vmax=None,
+                     cmap=None, cmap_set_under=None, cmap_scale='linear'):
+
+    """
+    TODO: description
+
+    Parameters
+    ----------
+    config : polaris.config.PolarisConfigParser
+        Configuration options with parameters used to construct the vertical
+        grid
+
+    ds : xarray.Dataset
+        A data set containing ``bottomDepth`` and ``ssh`` variables used to
+        construct the vertical coordinate
+    dsMesh : xarray.Dataset
+    """
+
+    try:
+        os.makedirs(os.path.dirname(outFileName))
+    except OSError:
+        pass
+
+    if os.path.exists(outFileName):
+        return
+
+    if title is None:
+        title = fieldName
+
+    for var in ['maxLevelCell']:
+        if var not in dsMesh:
+            raise ValueError(f'{var} must be added to dsMesh before plotting.')
+    if fieldName not in ds:
+        raise ValueError(f'{fieldName} must be present in ds before plotting.')
+
+    field = ds[fieldName]
+
+    if 'Time' in field.dims and tIndex is None:
+        tIndex = 0
+    if tIndex is not None:
+        field = field[tIndex, :]
+    if 'nVertLevels' in field.dims:
+        zIndex = 0
+    if zIndex is not None:
+        field = field[:, zIndex]
+
+    oceanMask = dsMesh.maxLevelCell - 1 >= 0
+    oceanMask = _remove_boundary_cells_from_mask(dsMesh, oceanMask)
+    oceanPatches = _compute_cell_patches(dsMesh, oceanMask)
+    oceanPatches.set_array(field[oceanMask])
+    if cmap is not None:
+        oceanPatches.set_cmap(cmap)
+    if cmap_set_under is not None:
+        current_cmap = oceanPatches.get_cmap()
+        current_cmap.set_under(cmap_set_under)
+
+    oceanPatches.set_edgecolor('face')
+    oceanPatches.set_clim(vmin=vmin, vmax=vmax)
+
+    if cmap_scale == 'log':
+        oceanPatches.set_norm(LogNorm(vmin=max(1e-10, vmin),
+                              vmax=vmax, clip=False))
+
+    width = dsMesh.xCell.max() - dsMesh.xCell.min()
+    length = dsMesh.yCell.max() - dsMesh.yCell.min()
+    aspect_ratio = width.values / length.values
+    fig_width = 4
+    legend_width = fig_width / 5
+    figsize = (fig_width + legend_width, fig_width / aspect_ratio)
+
+    plt.figure(figsize=figsize)
+    ax = plt.subplot(111)
+    ax.add_collection(oceanPatches)
+    plt.axis([0, 500, 0, 1000])
+    ax.set_aspect('equal')
+    ax.autoscale(tight=True)
+    plt.colorbar(oceanPatches, extend='both', shrink=0.7)
+    plt.title(title)
+    plt.tight_layout(pad=0.5)
+    plt.savefig(outFileName)
+    plt.close()
+
+
+def _remove_boundary_cells_from_mask(dsMesh, mask):
+    areaCell = dsMesh.areaCell.values
+    nVerticesOnCell = dsMesh.nEdgesOnCell.values
+    verticesOnCell = dsMesh.verticesOnCell.values - 1
```

Comment:
I'm on the fence about whether to allow these MPAS mesh variables to be mixed case.  It's maybe more intuitive if they match the MPAS names but it's a pretty fundamental no-no in python style.  I think I'm leaning toward using lowercase for them, too.

Reply:
Now changed to lowercase for MPAS variables